hashintel/hash
# Entity type editor 404 page + other UX fixes

## 🌟 What is the purpose of this PR?




Currently, when the entity type editor (either the new page, or the edit page) thinks you're on an invalid route, it attempts to redirect you somewhere sensible. This creates confusing code and is not necessary – we can just show a 404 page. This PR introduces that.

It also solves two problems where the page can jump about when publishing the first version of the entity type (see below). 

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203312852763953/1203457314114990/f) _(internal)_

## 🚫 Blocked by




N/A

## 🔍 What does this change?




- Default for `ownedById` for `useBlockProtocolCreateEntityType` is now `null` instead of an empty string. This improves code readability. 
- `useOrgs` and `useUsers` now have the option of using cache, which I've done for `useGetAccountIdForShortname` – I don't believe shortnames can change for a user or org, so this seems a sensible optimisation for now (makes going between the create page and the edit page much smoother) 
- Both the edit and new entity type pages show a 404 when the entity type or namespace have finished loading, but don't exist, instead of redirecting. 
- Removed old references to authenticatedUser, now that we work from the namespace instead
- The entity type editor now shows tabs row even when in draft mode, it just hides the other tabs. This prevents a weird jump when publishing the first version
- The edit bar now doesn't change props to the discard button as its animating away. This prevents a weird jump when publishing the first version
- Properly guard `generateEntityTypeBaseUriForUser` on existence of a namespace shortname

## 📜 Does this require a change to the docs?



N/A

## ⚠️ Known issues



N/A

## 🐾 Next steps

N/A

## 🛡 What tests cover this?



N/A

## ❓ How to test this?



1. Create a new entity type, publish the first version, and refresh
2. Ensure you're not redirected to `@undefined/types/entity-type/...`
3. Check that missing entity types / namespaces properly render 404s
